DateTimeMate_test.go

```diff
@@ -1,361 +1,25 @@
 package DateTimeMate
 
 import (
-	"fmt"
-	"github.com/golang-module/carbon/v2"
-	"strings"
 	"testing"
 )
 
-func testStartEnd(t *testing.T, start, end string, brief bool, correct string) {
-	diff := NewDiff(
-		DiffWithStart(start),
-		DiffWithEnd(end),
-		DiffWithBrief(brief))
-	result, _, err := diff.CalculateDiff()
-	if err != nil {
-		t.Error(err)
-	}
-	if result != correct {
-		t.Errorf("[computed: %v] != [correct: %v]", result, correct)
-	}
-}
-
-func testAddSubContains(t *testing.T, from, period, correctAdd, correctSub string) {
-	dur := NewDur(
-		DurWithFrom(from),
-		DurWithDur(period))
-	future, err := dur.Add()
-	if err != nil {
-		t.Error(err)
-	}
-	if !strings.Contains(future[0], correctAdd) {
-		t.Errorf("[from: %v] [computed: %v] does not contain: [correct: %v]", from, future, correctAdd)
-	}
-
-	past, err := dur.Sub()
-	if err != nil {
-		t.Error(err)
-	}
-	if !strings.Contains(past[0], correctSub) {
-		t.Errorf("[from: %v] [computed: %v] does not contain: [correct: %v]", from, past, correctSub)
-	}
-}
-
-func testAddSubOutputFormat(t *testing.T, from, period, outputFormat, correctAdd, correctSub string) {
-	dur := NewDur(
-		DurWithFrom(from),
-		DurWithDur(period),
-		DurWithOutputFormat(outputFormat))
-	future, err := dur.Add()
-	if err != nil {
-		t.Error(err)
-	}
-	if !strings.Contains(future[0], correctAdd) {
-		t.Errorf("\n[from: %v]\n[period: %v]\n[computed: %v] does not contain:\n[correct: %v]", from, period, future[0], correctAdd)
-	}
-	return
-
-	past, err := dur.Sub()
-	if err != nil {
-		t.Error(err)
-	}
-	if !strings.Contains(past[0], correctSub) {
-		t.Errorf("[from: %v] [computed: %v] does not contain: [correct: %v]", from, past, correctSub)
-	}
-}
-
-func testAddSubWithRepeat(t *testing.T, from, period string, correctAdd, correctSub []string, repeat int) {
-	dur := NewDur(
-		DurWithFrom(from),
-		DurWithDur(period),
-		DurWithRepeat(repeat))
-	future, err := dur.Add()
-	if err != nil {
-		t.Error(err)
-	}
-	for i := range len(future) {
-		if !strings.Contains(future[i], correctAdd[i]) {
-			t.Errorf("[from: %v] [computed: %v] does not contain: [correct: %v]", from, future[i], correctAdd[i])
-		}
-	}
+func TestShrinkPeriod(t *testing.T) {
+	var allLongPeriods = []string{"nanoseconds", "microseconds", "milliseconds", "seconds", "minutes", "hours", "days", "weeks", "months", "years"}
+	var allShortPeriods = []string{"ns", "us", "ms", "s", "m", "h", "D", "W", "M", "Y"}
```

Comment:
You almost have everything to built the [map](https://github.com/jftuga/DateTimeMate/pull/5#discussion_r1683863526) here

Reply:
* fixed by using `abbrevMap`
* please review changes in the next PR.